### PR TITLE
Project manage membership fixes

### LINKED ISSUE #103: Project manage membership: Should not be able to associate org inbuilt role to users and groups
### Expected vs actual behavior

- Expected: Admins should be able to associate user to project / namespace scoped roles from Project manage membership -> Assign user to Project
- Actual: Choosen user selection gets cleared and also nothing happens on clicking Save & Exit  when an organization scoped role is chosen

### Steps to reproduce the bug

1. Login as Admin to Paralus Dashboard
2. Navigate to Project Manage Membership -> Assign User To Project
3. Choose a user and select Organization admin or Organization read only role
4. Observe the selected user dropdown and also try clicking Save & Exit

### Are you using the latest version of the project?

- Yes, v.0.1.0

### What is your environment setup? Please tell us your cloud provider, operating system, and include the output of `kubectl version --output=yaml` and `helm version`. Any other information that you have, eg. logs and custom values, is highly appreciated!

- Kind

### _(optional)_ If you have ideas on why the bug happens or how it can be solved, please provide it here
- NA

- [x] I've described the bug, included steps to reproduce it, and included my environment setup with all customizations.
- [x] I'm using the latest version of the project.

### PR DESCRIPTION
**Pull request by niravparikh05**
### What does this PR change?

- Fixes #103 #104 

### Does the PR depend on any other PRs or Issues? If yes, please list them.

- No

### Checklist

I confirm, that I have...

- [x] Read and followed the contributing guide in `CONTRIBUTING.md`
- [ ] Added tests for this PR
- [x] Formatted the code using `npm run format` (if applicable)
- [ ] Updated [documentation on the Paralus docs site](https://github.com/paralus/website/blob/main/docs) (if applicable)
- [x] Updated `CHANGELOG.md`

### CODE REVIEW

**Comment by meain** [comment]
Deleting a role after assigning seems to cause issues and does not seem to get deleted.
```
TypeError
data.metadata is undefined
Call Stack
 editProjectWithCallback/<
  actions/Projects.js:229:114
 middleware/
  node_modules/redux-thunk/es/index.js:17:18
 routerMiddleware/
  node_modules/connected-react-router/esm/middleware.js:28:18
 _loop/boundActionCreators[key]
  node_modules/react-redux/es/utils/bindActionCreators.js:13:16
 AssignedUsers/<
  appMain/routes/project/routes/ProjectDetail/AssignedUsers.js:248:30
 handleClick
  appMain/routes/project/routes/ProjectDetail/AssignedUsers.js:291:26
 handleConfirm
  components/DeleteIconComponent.js:120:12
 callCallback
  node_modules/react-dom/cjs/react-dom.development.js:3945:14
 invokeGuardedCallbackDev
  node_modules/react-dom/cjs/react-dom.development.js:3994:16
 invokeGuardedCallback
  node_modules/react-dom/cjs/react-dom.development.js:4056:31
```

**Comment by niravparikh05** [comment]
> Deleting a role after assigning seems to cause issues and does not seem to get deleted.
> 
> ```
> TypeError
> data.metadata is undefined
> Call Stack
>  editProjectWithCallback/<
>   actions/Projects.js:229:114
>  middleware/
>   node_modules/redux-thunk/es/index.js:17:18
>  routerMiddleware/
>   node_modules/connected-react-router/esm/middleware.js:28:18
>  _loop/boundActionCreators[key]
>   node_modules/react-redux/es/utils/bindActionCreators.js:13:16
>  AssignedUsers/<
>   appMain/routes/project/routes/ProjectDetail/AssignedUsers.js:248:30
>  handleClick
>   appMain/routes/project/routes/ProjectDetail/AssignedUsers.js:291:26
>  handleConfirm
>   components/DeleteIconComponent.js:120:12
>  callCallback
>   node_modules/react-dom/cjs/react-dom.development.js:3945:14
>  invokeGuardedCallbackDev
>   node_modules/react-dom/cjs/react-dom.development.js:3994:16
>  invokeGuardedCallback
>   node_modules/react-dom/cjs/react-dom.development.js:4056:31
> ```

Thank you @meain for reporting this, that would have just slipped in if not. I have fixed it and pushed the changes in.

File: src/appMain/routes/project/routes/AssignGroup/ProjectRoleWidget/components/RolesCard.js [hunk 1]
```diff
@@ -38,45 +38,42 @@ const RolesCard = ({
...
+                  role="listitem"
+                  button
+                  onClick={handleToggle(value)}
+                  className={classes.listItem}
+                >
+                  <ListItemIcon>
+                    <Checkbox
+                      color="primary"
+                      checked={checked.indexOf(value) !== -1}
+                      tabIndex={-1}
```
Comment (meain) [inline src/appMain/routes/project/routes/AssignGroup/ProjectRoleWidget/components/RolesCard.js:60]: Neat addition.

File: src/appMain/routes/project/routes/AssignGroup/ProjectRoleWidget/index.js [hunk 2]
```diff
@@ -85,6 +87,22 @@ const ProjectRoleWidget = ({
...
+        }
+      }
+      if (value.metadata.name === "NAMESPACE_READ_ONLY") {
+        const pai = checked.findIndex(
+          (element) => element.metadata.name === "NAMESPACE_ADMIN"
+        );
+        if (pai !== -1) {
+          newChecked.splice(pai, 1);
+        }
+      }
```
Comment (meain) [inline src/appMain/routes/project/routes/AssignGroup/ProjectRoleWidget/index.js:105]: Can we simplify the whole things with something like:
```javascript
const pairs = ["PROJECT", "INFRA", "NAMESPACE"]
  .map((e) => [
    [e + "_ADMIN", e + "_READ_ONLY"],
    [e + "_READ_ONLY", e + "_ADMIN"],
  ])
  .reduce((a, b) => [...a, ...b], []);

for (let pair of pairs) {
  if (value.metadata.name === pair[0]) {
    const item = checked.findIndex(
      (element) => element.metadata.name === pair[1]
    );
    if (item !== -1) newChecked.splice(item, 1);
  }
}
```

**Review by meain — COMMENTED** [review]
Some tiny nits, but looks good otherwise.

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,8 @@ All notable changes to this project will be documented in this file.
 - Fix blank screen on initial login from [meain](https://github.com/meain)
 - Fix time/date rendering in audit logs from [meain](https://github.com/meain)
 - Invalid date displayed on cluster page [niravparikh05](https://github.com/niravparikh05)
+- Project manage membership: Should not be able to associate org role to users and groups [niravparikh05](https://github.com/niravparikh05)
+- Project manage membership: Unable to associate inbuilt namespaced roles to users and groups [niravparikh05](https://github.com/niravparikh05)
```
Comment (meain) [inline CHANGELOG.md:15]:
```suggestion
- Project manage membership: Should not be able to associate org role to users and groups from [niravparikh05](https://github.com/niravparikh05)
- Project manage membership: Unable to associate inbuilt namespaced roles to users and groups from [niravparikh05](https://github.com/niravparikh05)
```

Comment (meain) [reply CHANGELOG.md:15]: Maybe even update the one before.

File: src/appMain/routes/project/routes/AssignGroup/ProjectRoleWidget/index.js (hunk 2 above)
Comment (niravparikh05) [reply src/appMain/routes/project/routes/AssignGroup/ProjectRoleWidget/index.js]: this is neat, have refactored all files applicable with above code.

File: CHANGELOG.md (hunk 3 above)
Comment (niravparikh05) [reply CHANGELOG.md]: addressed

**Review by meain — APPROVED** [review]

**Pull request merged**